standalone/src/configuration/xml_configuration_parsing.cpp

```diff
@@ -66,38 +170,89 @@ ZoneSet parseZoneSet(const tinyxml2::XMLElement* xml_set_element)
+      set.warn2_Sub1_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
     }
     else if (textIsEqual(xml_set_detail_type_element, "muting1"))
     {
-      set.muting1_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
+      set.muting1_Sub1_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
     }
     else if (textIsEqual(xml_set_detail_type_element, "muting2"))
     {
-      set.muting2_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
+      set.muting2_Sub1_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
     }
     else
     {
       throw XMLConfigurationParserException("Could not parse. Invalid <type> must be \"roOSSD1\", \"roOSSD2\", "
                                             "\"roOSSD3\", \"warn1\", \"warn2\", \"muting1\" or \"muting2\".");
     }
+    // Move to next <zoneSetDetail>
+    xml_set_detail_element = xml_set_detail_element->NextSiblingElement("zoneSetDetail");
+  }
+
+  // Set default resolution for now this is only known implicitly
+  set.resolution_ = DEFAULT_ZONESET_ANGLE_STEP;
+
+  return set;
+}
+
+ZoneSet parseZoneSetSub2(const tinyxml2::XMLElement* xml_set_element, ZoneSet set)
+{
+  const tinyxml2::XMLElement* xml_set_detail_element = getFirstChildElement(xml_set_element, "zoneSetDetail");
 
+  while (xml_set_detail_element)
+  {
+    const tinyxml2::XMLElement* xml_set_detail_type_element = getFirstChildElement(xml_set_detail_element, "type");
+    const tinyxml2::XMLElement* xml_set_detail_ro_element = getFirstChildElement(xml_set_detail_element, "ro");
+
+    if (textIsEqual(xml_set_detail_type_element, "roOSSD1"))
+    {
+      set.safety1_Sub2_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
+    }
+    else if (textIsEqual(xml_set_detail_type_element, "roOSSD2"))
+    {
+      set.safety2_Sub2_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
+    }
+    else if (textIsEqual(xml_set_detail_type_element, "roOSSD3"))
+    {
+      set.safety3_Sub2_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
+    }
+    else if (textIsEqual(xml_set_detail_type_element, "warn1"))
+    {
+      set.warn1_Sub2_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
+    }
+    else if (textIsEqual(xml_set_detail_type_element, "warn2"))
+    {
+      set.warn2_Sub2_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
+    }
+    else if (textIsEqual(xml_set_detail_type_element, "muting1"))
+    {
+      set.muting1_Sub2_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
+    }
+    else if (textIsEqual(xml_set_detail_type_element, "muting2"))
+    {
+      set.muting2_Sub2_ = ro_string_to_vec(getText(xml_set_detail_ro_element));
+    }
+    else
+    {
+      throw XMLConfigurationParserException("Could not parse. Invalid <type> must be \"roOSSD1\", \"roOSSD2\", "
+                                            "\"roOSSD3\", \"warn1\", \"warn2\", \"muting1\" or \"muting2\".");
+    }
```

Comment:
It seems like much code duplication. I assume you did consider and there is no easy way to avoid it?